### PR TITLE
#4785 Update date-picker to properly set initialValue

### LINKED ISSUE #4785: Vaccination event Edit model: Date-picker and date is not updating correctly
## Describe the bug

- The date that come in to the form is in the wrong format, and the Date-picker, cannot handle it. It does not bug out but the interface shown would always default to the latest date. 
- Also saving the date has no effect on listing page.

### To reproduce

Steps to reproduce the behaviour:

https://user-images.githubusercontent.com/683452/177276639-d5fa6c1f-af52-4b69-a5e9-997fefdf5a6d.mp4

### Expected behaviour

- Datepicker should show the selected date not today's date.
- The format in form and format after save should be same.
- The saved date should show up in listing page.

### Proposed Solution

- Update Datepicker component of JSONForm, now this is generic component and would affect many forms so must be careful during coding and testing. But I believe the component itself needs to fixed otherwise this issue will show up in many places. 

### Version and device info

- App version:
- Tablet model:
- OS version:

### Additional context

Related to (bug fix for) https://github.com/openmsupply/mobile/issues/4715

### PR DESCRIPTION
**Pull request by sworup**
Fixes #4785

## Change summary

- Previously the datepicker, when pulling date from sever, was getting the date in wrong format, saving this date, if the date was converted to dd/mm/yyyy format would show wrong date.
- if you click the datepicker button, the date selected would be today's date not the date in input box.
- If you try to edit the date in input box mid-way or make it blank, and open datepicker button the app would error out.
- if you add date in correct format and save, the saved date would be in wrong format, that is day and month swaped.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Go to `Vaccinations` page
- [ ] Select vaccination history of one of the patient from online search
- [ ] see the date of vaccination
- [ ] Select to edit details
- [ ] In pages additional details see that the date is in dd/mm/yyyy format. 
- [ ] click the datepicker button, the date should show properly in datepicker.
- [ ] now in the date input box, press backspace to delete the date, it should not error out or say invalid date
- [ ] click the datepicker button again, and the picker now should show today's date and let you select other dates

### Related areas to think about

This datepicker is not limited to vaccination event details page so we have to look at other pages where this datepicker is used too.
Save or edit might bug out by taking invalid date if this code has bugs so maybe test edit and save too.

### CODE REVIEW

**Comment by sworup** [comment]
This is still in draft.
There is a problem where after update vax event listing will still show older date.
Working on it.

**Comment by sworup** [comment]
@chastichno this is ready to review, it fixes the datepicker issue, other bugs I would create other issues to fix.

**Review by chastichno — CHANGES_REQUESTED** [review]
@sworup With the changes made in this PR users are no longer able to enter the Date from the keyboard

File: src/widgets/JSONForm/widgets/DatePicker.js [hunk 1]
```diff
@@ -21,11 +21,23 @@ export const DatePicker = ({
...
+  const [selectedDate, setSelectedDate] = useState(null);
+
   const handleChange = dateString => {
     onChange(dateString);
     onBlur(id, dateString);
   };
 
+  React.useEffect(() => {
+    const alternateFormatDate = moment(value, 'YYYY-MM-DD', true);
+    const expectedFormateDate = alternateFormatDate.isValid()
```
[the remark this answers is not in the document]
Comment (sworup) [reply src/widgets/JSONForm/widgets/DatePicker.js]: Resolved with e9afe902

**Review by chastichno — CHANGES_REQUESTED** [review]
When I manually erase the date form the input field I get a Warning:
Screen Shot 2022-07-20 at 12 17 40 PM

File: src/widgets/JSONForm/widgets/DatePicker.js [hunk 2]
```diff
@@ -21,11 +21,23 @@ export const DatePicker = ({
 }) => {
   const { focusController } = useJSONFormOptions();
   const ref = focusController.useRegisteredRef();
+
+  const [selectedDate, setSelectedDate] = useState('');
```
Comment (chastichno) [inline src/widgets/JSONForm/widgets/DatePicker.js:25]:
```suggestion
  const [selectedDate, setSelectedDate] = useState(Date());
```

Comment (sworup) [reply src/widgets/JSONForm/widgets/DatePicker.js]: Resolve with b0d9bf68

**Review by chastichno — APPROVED** [review]
Looks good to me! 🙌✨

**Pull request merged**